Documentation for memory optimization and micro iterations topics (#1264)
AntaresSimulatorTeam/antares-xpansion: docs/user-guide/get-started/options-definition.md

Reviewer: ```suggestion
# Settings for launching the Benders executable
```

Reviewer: ```suggestion
The execution of the solver is configured through an input file named `options.json`, which must be supplied at runtime. This JSON configuration file defines all the parameters required to control the behavior of the solver. It typically includes the solver’s operational settings, algorithmic options, and problem-specific configurations. By adjusting the attributes in this file, users can fine-tune aspects such as decomposition settings, convergence criteria, logging behavior, performance-related options, and input/output paths.

The executable can be launched with `benders options.json`
```

|OUTPUTROOT | `.` |  Path to the folder where output files should be printed  | `boolean` |
|OUTER_LOOP_OPTION_FILE | `adequacy_criterion.yml` |  Outer Loop Options file  |`string`|
|AREA_FILE | `area.txt` |  Area file used to get areas on which external criteria (LOLD, UnsuppliedEnergy) are computed |`string` |

> |CACHE_PROBLEMS | `1` |  Subproblem caching mode: `1` = all problems loaded in memory (classic mode), `2` = subproblems stored in disk cache rather than loaded in memory (reduces RAM usage at the expense of a slight CPU time performance reduction), `3` = memory-optimized mode using a shared skeleton problem with per-subproblem CSV data  | `1`, `2` or `3` |

Reviewer: This starts at 1, not 0 ?

Reviewer: Fix that !

> |CACHE_PROBLEMS | `1` |  Subproblem caching mode: `1` = all problems loaded in memory (classic mode), `2` = subproblems stored in disk cache rather than loaded in memory (reduces RAM usage at the expense of a slight CPU time performance reduction), `3` = memory-optimized mode using a shared skeleton problem with per-subproblem CSV data  | `1`, `2` or `3` |

Reviewer: ```suggestion
|CACHE_PROBLEMS | `1` |  Subproblem caching mode: `1` = all problems loaded in memory (classic mode), `2` = subproblems stored in disk cache rather than loaded in memory (reduces RAM usage at the expense of a slight CPU time performance reduction), `3` = in-memory compact mode using a shared skeleton problem with per-subproblem RHS and objective coefficients data  | `1`, `2` or `3` |
```
